tools/autograd/gen_autograd.py

```diff
@@ -64,11 +64,23 @@
     'sparse_coo_tensor_with_dims_and_tensors': 'values',
 }
 
+# Functions for which we use CreationMeta::MULTI_OUTPUT_SAFE. I.e., the ones for
+# which inplace modification of otputs is being gradually deprecated.
+MULTI_OUTPUT_SAFE_FUNCTIONS = {
+    'split',
+    'split_with_sizes',
+    'chunk',
+}
+
 # note: some VIEW_FUNCTIONS are just compositions of the view functions above
 # this list contains both the root view functions and any that are purely composed
 # of viewing functions, and is used by the JIT to determine when an operator
 # returns a view of its inputs
-RETURNS_VIEWS_OF_INPUT = set(VIEW_FUNCTIONS.keys()).union({'chunk', 'split'})
+RETURNS_VIEWS_OF_INPUT = set(VIEW_FUNCTIONS.keys()).union({
+    'chunk',
+    'split',
```

Comment:
There is this comment above:
```
# note: some VIEW_FUNCTIONS are just compositions of the view functions above                            
# this list contains both the root view functions and any that are purely composed                       
# of viewing functions, and is used by the JIT to determine when an operator                             
# returns a view of its inputs
```

and in `aten/src/ATen/native/TensorShape.cpp` you can see that `split` calls `Tensor::narrow` so I thought that `self` belongs in `RETURNS_VIEWS_OF_INPUT` rather than `VIEW_FUNCTIONS`. But I must admit that I don't fully understand what's going on here.

Reply:
The actual implementation is "unrelated" here.
The short answer is that we have 2 ways to call a low level function:
- As a composite op where we have a single generatic implementation that calls into other ops
- As a non-composite op for which the dispatcher is responsible to call the right implementation.

These match the `elementary` vs `non-elementary` I mentioned above.
Finding which class each function belong to is a bit tricky though but in general:
- If there is an entry in derivatives.yaml, it is not composite
- If there are special type dispatch in native_functions.yaml, it is not composite

To be 100% sure, I usually check the `torch/csrc/autograd/generated/VariableTypeEverything.cpp` file and find my function in there. If the inner call is wrapped into `AutoNonVariableTypeMode` then it is not composite, otherwise it is.


For the case of `split`, you can see that it has an entry in `derivatives.yaml` so it should be in the first list.

Reply:
Thanks for the explanation. I will need to read it a few more times to understand, but your remark about  `torch/csrc/autograd/generated/VariableTypeEverything.cpp` provides easy enough rule for what to do here.